src/foam/parse/FScriptParser.java

```diff
@@ -189,7 +192,83 @@ public Object value() {
         ( Expr ) values[2] : new foam.mlang.Constant (values[2]));
       return bin;
     });
+    grammar.addSymbol(
+      "FORMULA",
+      new Repeat(grammar.sym("MINUS"), Literal.create("+"),1)
+    );
+    grammar.addAction("FORMULA", (val, x) -> {
+      Object[] vals = (Object[]) val;
+      if ( vals[0] == null ) return null;
+      List<Expr> args = new ArrayList();
```

Comment:
new ArrayList(vars.length) is more efficient because it avoids wasting space or having to grow the array. But, Why do we need this to be an ArrayList in the first place if we just convert it to an Expr[] below? Couldn't we just create the Expr[] of the correct size?